### PR TITLE
UI-66 Build accordion component

### PR DESCRIPTION
**Pull request by tmoran18**
Created Accordion component using rc-collapse package - https://www.npmjs.com/package/rc-collapse/v/3.1.0

**IE 10 Testing**
Screen Shot 2021-03-15 at 11 02 50 am

**Story GIF**
Accordion

**Mobile**
Screen Shot 2021-03-15 at 11 16 47 am

### CODE REVIEW

File: src/Accordion/Accordion.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import React from "react";
```
Comment (MichaelDorkhom) [inline src/Accordion/Accordion.stories.tsx:1]: Can you change all double quotes to single quotes

File: src/Accordion/Accordion.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import { storiesOf } from "@storybook/react";
+import { Accordion } from "./Accordion";
```
Comment (MichaelDorkhom) [inline src/Accordion/Accordion.stories.tsx:3]: we structure imports in 3 separate blocks (separated by a blank line): external library imports, internal imports and style imports in that order

File: src/Accordion/Accordion.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+storiesOf("Components.Accordion", module).add("Accordion", () => {
+  return (
+    <div style={{ maxWidth: "700px" }}>
+      <Accordion>
+        {data.map((item) => {
+          return (
+            <AccordionItem key={item.header} header={item.header}>
+              <div>{item.body}</div>
+            </AccordionItem>
+          );
```
Comment (MichaelDorkhom) [inline src/Accordion/Accordion.stories.tsx:38]: semicolon not needed

File: src/Accordion/Accordion.tsx [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+          stroke-width="3"
+        ></path>
+      </svg>
+    );
+  };
+
+  return <Collapse expandIcon={expandIcon}>{children}</Collapse>;
+};
+
+export { Accordion };
```
Comment (MichaelDorkhom) [inline src/Accordion/Accordion.tsx:40]: can you remove all semicolons when not required

File: src/Accordion/index.scss [hunk 5]
```diff
@@ -0,0 +1,20 @@
+@import "../style/colors.scss";
+
+  .rc-collapse-content {
```
Comment (MichaelDorkhom) [inline src/Accordion/index.scss:3]: can you fix the indentation

File: src/Accordion/index.scss [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+  
+  .rc-collapse-content-inactive {
+    display: none;
+  }
+  
+  .rc-collapse {
+    position:relative;
+  }
+
+  svg {
```
Comment (MichaelDorkhom) [inline src/Accordion/index.scss:15]: you'll need to make this more specific, so it doesn't get applied to all svgs across the codebase

File: src/AccordionItem/AccordionItem.tsx [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+import "./index.scss";
+
+interface Iprops {
+  header: string;
+}
+
+const AccordionItem: React.FC<Iprops> = (props) => {
+  const { header } = props;
+
+  return <Panel {...props} showArrow={true} header={header}></Panel>;
```
Comment (MichaelDorkhom) [inline src/AccordionItem/AccordionItem.tsx:13]: You can remove `header={header}` if you're passing `{...props}`

File: src/AccordionItem/AccordionItem.tsx [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+import { Panel } from "rc-collapse";
+
+import "./index.scss";
+
+interface Iprops {
+  header: string;
+}
+
+const AccordionItem: React.FC<Iprops> = (props) => {
+  const { header } = props;
```
Comment (MichaelDorkhom) [inline src/AccordionItem/AccordionItem.tsx:11]: not needed if you're passing `{...props}` to `Panel`

File: src/AccordionItem/index.scss [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+      text-decoration: underline;
+    }
+  }
+}
+
+.rc-collapse-content {
+  background: white;
+  padding:15px;
+}
+
```
Comment (MichaelDorkhom) [inline src/AccordionItem/index.scss:34]: extra blank line

File: src/Accordion/Accordion.tsx [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+import Collapse from "rc-collapse";
+
+import "./index.scss";
+
+interface IAccordionActive {
+  isActive?: boolean;
+}
+
+const Accordion: React.FC = ({ children }) => {
+  const expandIcon = ({ isActive }: IAccordionActive) => {
```
Comment (MichaelDorkhom) [inline src/Accordion/Accordion.tsx:11]: Can you make this a reusable Icon, and move it under the "Icons" folder (e.g. `src/Icons/CollapseArrow.tsx`)

File: src/AccordionItem/AccordionItem.tsx (hunk 7 above)
Comment (MichaelDorkhom) [reply src/AccordionItem/AccordionItem.tsx:13]: however, passing `{...props}` is never ideal. You want to have control and what attributes are being used, so it's always better to explicitly pass the ones we are using instead

File: src/AccordionItem/index.scss [hunk 11]
```diff
@@ -0,0 +1,34 @@
+@import "../style/colors.scss";
+
+.rc-collapse > .rc-collapse-item > .rc-collapse-header {
+  padding: 10px 15px;
+  color: #666;
```
Comment (MichaelDorkhom) [inline src/AccordionItem/index.scss:5]: can this use one of the colours we already have defined

File: src/AccordionItem/AccordionItem.tsx [hunk 12]
```diff
@@ -0,0 +1,16 @@
+import React from "react";
+import { Panel } from "rc-collapse";
+
+import "./index.scss";
+
+interface Iprops {
```
Comment (MichaelDorkhom) [inline src/AccordionItem/AccordionItem.tsx:6]: can you update this to `IProps`. Convention is to start interfaces with `I` and types with `T`, then add the name in pascal case. i.e.

`TThisIsANewType` and `IOrAnInterface`

File: src/Accordion/Accordion.stories.tsx (hunk 1 above)
Comment (MichaelDorkhom) [reply src/Accordion/Accordion.stories.tsx:1]: this is something you can add to your IDE settings if these are automaticallyy imported

File: src/AccordionItem/AccordionItem.tsx (hunk 7 above)
Comment (julianjuko) [reply src/AccordionItem/AccordionItem.tsx:13]: +1 for not using `...props` and defining all props explicitly

File: src/AccordionItem/index.scss (hunk 9 above)
Comment (julianjuko) [reply src/AccordionItem/index.scss:34]: @tmoran18 rule of thumb, CSS files don't have a blank line at the end, TS and TSX do (along with most other filetypes)

**Review by julianjuko — APPROVED** [review]
lgtm, just fix up @MichaelDorkhom comments. Nice one!

File: src/Accordion/index.scss [hunk 13]
```diff
@@ -0,0 +1,19 @@
...
+  overflow: hidden;
+}
+
+.rc-collapse-content-inactive {
+  display: none;
+}
+
+.rc-collapse {
+  position: relative;
+  & svg {
```
Comment (MichaelDorkhom) [inline src/Accordion/index.scss:13]: you don't need `&` if it's already nested

**Review by MichaelDorkhom — APPROVED** [review]

**Pull request merged**